docker/Dockerfile

```diff
@@ -47,14 +48,14 @@ RUN /rgtm/pre141/bin/pip install -U pip \
       rospkg==1.2.3 \
       distribute==0.7.3 \
       pyyaml==5.3 \
-      vcstools==0.1.30 \
+      vcstools==0.1.38 \
```

Comment:
I haven't tested this, but I'm not sure we can do this. The whole things is rather dependend on specific versions.

Have you tested the image with this version of `vcstools`?

Reply:
When I execute the example from the readme without resolving the version conflicts, I get this error:

Requested timepoint: '2017-03-01T08:57:20Z' (1488358640)
Resetting local rosdistro clone ..
Previous HEAD position was 2af311e20 jsk_planning: 0.1.9-0 in 'kinetic/distribution.yaml' [bloom]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Determined rosdistro commit: 2af311e2 (authored: 1488354358)
Reverting to ros/rosdistro@2af311e2
Cache already exists for (distro; stamp) tuple, skipping generation
Updating local rosdistro index.yaml to use cache from the past ..
Invoking: rosinstall_generator --rosdistro=kinetic kobuki_ftdi --deps --deps-only --tar
Using ROS_DISTRO: kinetic
/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/__init__.py:102: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(yaml_str)
/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/__init__.py:170: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(yaml_str)
Traceback (most recent call last):
  File "/rgtm/post141/bin/rosinstall_generator", line 8, in <module>
    sys.exit(main())
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/cli.py", line 163, in main
    upstream_version_tag=args.upstream, upstream_source_version=args.upstream_development)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/generator.py", line 364, in generate_rosinstall
    limit_depth=deps_depth, source=upstream_source_version)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosinstall_generator/distro.py", line 91, in get_recursive_dependencies
    ignore_pkgs=dependencies | excludes, limit_depth=limit_depth)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 78, in get_recursive_depends
    deps = self.get_depends(next_pkg_to_check, depend_type, ros_packages_only=ros_packages_only)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 61, in get_depends
    deps = self._get_dependencies(pkg_name, depend_type)
  File "/rgtm/post141/local/lib/python2.7/site-packages/rosdistro/dependency_walker.py", line 126, in _get_dependencies
    return set([d.name for d in deps[dep_type]])
KeyError: 'buildtool_export'

Reply:
I know, that's #17.

(could you also please use [Github codeblocks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) for console copy-pastes?)

Reply:
The docker image built from the dockerfile in this PR executes successfully and creates this output: 

- tar:
    local-name: catkin
    uri: https://github.com/ros-gbp/catkin-release/archive/release/kinetic/catkin/0.7.6-0.tar.gz
    version: catkin-release-release-kinetic-catkin-0.7.6-0
- tar:
    local-name: ecl_core/ecl_command_line
    uri: https://github.com/yujinrobot-release/ecl_core-release/archive/release/kinetic/ecl_command_line/0.61.17-0.tar.gz
    version: ecl_core-release-release-kinetic-ecl_command_line-0.61.17-0
- tar:
    local-name: ecl_tools/ecl_license
    uri: https://github.com/yujinrobot-release/ecl_tools-release/archive/release/kinetic/ecl_license/0.61.6-0.tar.gz
    version: ecl_tools-release-release-kinetic-ecl_license-0.61.6-0

Reply:
I'm willing to accept `wstool` can be updated, but without https://github.com/ros-infrastructure/rosdistro/compare/master...rosin-project:rosin_bughunt_0.6.8, you're definitely going to run into problems later.

Reply:
So the solution would be to enforce an older version of `rosinstall_generator` then?

Reply:
We're conflating things here I believe.

`vcstools` can probably updated, it's basically a stand-alone tool.

`rosdistro` needs a patch, otherwise the time-machine can't work.

The complicating factor here is that I believe #17 can't be fixed unless you update one of the dependencies (probably `rosdistro`), which means the patch would have to be rebased. But it's unclear to me which version of `rosdistro` would be needed.

(also: this should be a comment on the main PR, not on this specific line in the patch)